src/java.desktop/share/native/libawt/awt/image/awt_parseImage.c

```diff
@@ -97,9 +97,11 @@ int awt_parseImage(JNIEnv *env, jobject jimage, BufImageS_t **imagePP,
     }
 
     /* Parse the color model */
+    ColorModelS_t *cmP = &imageP->cmodel;
     if ((status = awt_parseColorModel(env, jcmodel, imageP->imageType,
-                                      &imageP->cmodel)) <= 0) {
+                                      cmP)) <= 0) {
         awt_freeParsedRaster(&imageP->raster, FALSE);
+        if (cmP->nBits != NULL) free(cmP->nBits);
```

Comment:
It seems wrong to me that awt_parseColorModel should return a failure code and not have cleaned up any storage it itself allocated.
In other words this clean up belongs in there.

Reply:
Hi Phil, I moved the cleanup.

Another question - what do you think about introducting a macro JNU_CHECK_EXCEPTION_FREE_RETURN ? We have quite a number of places using this pattern (exception check , free and return)  , in this change now as well.  Might be worth having a macro for this.